backend/middlewares/authentication.js

```diff
@@ -0,0 +1,31 @@
+const jwt = require('jsonwebtoken');
+
+class Authentication {
+
+    //TODO: in ../config/config.js, readFileSync(file with privateKey) ?
+    static privateKey() {
+        return 'PRIVATE_KEY';
+    }
+
+    static generateAuthToken(user) {
+        try {
+            const authToken = jwt.sign({ data: JSON.stringify(user) }, this.privateKey(), { 
+                expiresIn: '1d',
+                algorithm: 'HS256'
+            });
+            return authToken;
+        } catch (error) {
+            throw error
+        }
+    }
+
+    static extractUser(authToken) {
+        try {
```

Comment:
Since the frontend will be sending an authToken as a Bearer token it's going to be in the form of `Bearer {jwt}` so maybe in extractUser() you could first parse the authToken by checking if it starts with "Bearer" and then decoding the jwt next to it

Reply:
yes thanks